src/py/kaleido/kaleido.py

```diff
@@ -313,7 +313,7 @@ async def calc_fig(
         `write_fig` does, although its arguments are a subset of those of `write_fig`.
         This function is currently just meant to bridge the old and new API.
         """
-        if not hasattr(fig, "to_dict") or isinstance(fig, Iterable):
+        if not hasattr(fig, "to_dict") and isinstance(fig, Iterable):
```

Comment:
@ayjayt If the goal is to check that the input is the right type, how about something like

```suggestion
        if not isinstance(fig, dict) or not hasattr(fig, "to_dict"):
            raise TypeError("`fig` must be a dictionary or a Plotly Figure object")
```

Reply:
Yeah I think that makes sense for the PR related to https://github.com/plotly/Kaleido/issues/270

I think that PR will be this PR, but I've just been working on https://github.com/plotly/Kaleido/issues/269 because its a bit easier

Reply:
I was also thinking of add a `hasattr(fig, "data")` to verify the dictionary is a properly structured dictionary